File: 3. roman-to-integer/roman-to-integer.py
def romanToInt(s: str) -> int:
    rome = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
    numLst = list(s.upper())

    sum = 0
    count = 0
    running = True

    while (running):
        if (count == len(numLst) - 1):
            sum = sum + rome[numLst[count]]
            running = False
        elif (rome[numLst[count]] < rome[numLst[count+1]]):
            sum = sum + rome[numLst[count+1]] - rome[numLst[count]]
            count = count + 1
            if (count == len(numLst) - 1):
                running = False
            count = count + 1
        else:
            sum = sum + rome[numLst[count]]
            count = count + 1

    return(sum)


print(romanToInt("XIV"))

File: 3. roman-to-integer/roman-to-integer.php
<?php

function romanToInt($s) {
    $rome = array("I" => 1, "V" => 5, "X" => 10, "L" => 50, "C" => 100, "D" => 500, "M" => 1000);
    $numLst = str_split(strtoupper($s), 1);

    $sum = 0;
    $cnt = 0;
    $running = true;

    while ($running) {
        if ($cnt == count($numLst) - 1) {
            $sum += $rome[$numLst[$cnt]];
            $running = false;
        } elseif ($rome[$numLst[$cnt]] < $rome[$numLst[$cnt + 1]]) {
            $sum += $rome[$numLst[$cnt + 1]] - $rome[$numLst[$cnt]];
            $cnt++;
            if ($cnt == count($numLst) - 1) {
                $running = false;
            }
            $cnt++;
        } else {
            $sum += $rome[$numLst[$cnt]];
            $cnt++;
        }
    }

    return($sum);
}

echo(romanToInt("XIV"));
echo("\n");